scripts/PearlPowderISIS/pearl_cycle_factory.py

```diff
@@ -0,0 +1,154 @@
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 78610):
+        cycle = "12_4"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 80073):
+        cycle = "12_5"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 80812):
+        cycle = "13_1"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 81280):
+        cycle = "13_2"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 82317):
+        cycle = "13_3"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 82764):
+        cycle = "13_4"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 84841):
+        cycle = "13_5"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 86631):
+        cycle = "14_1"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 88087):
+        cycle = "14_2"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 89389):
+        cycle = "14_3"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 90481):
+        cycle = "15_1"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 91530):
+        cycle = "15_2"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(datadir)
+    elif (runno < 92432):
+        cycle = "15_3"
+        print "cycle is set to", cycle
+        instver = "new2"
+        datadir = "X:\data\cycle_" + cycle + "\\"
+        pearl_routines.PEARL_setdatadir(currentdatadir)
```

Comment:
@FedeMPouzols they seem to use `currentdatadir` more recently raw directories entries unless they have been using `datadir` above.. and yes the script in **To Test** should work without going on to network. I have got all the raw files needed to run the script in `\\olympic\Babylon5\Public\Shahroz\PEARL\RawFiles`